### PR TITLE
Draft version updates and changelog

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

File: kvdb-rocksdb/CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 
-The format is based on [Keep a Changelog]. 
+The format is based on [Keep a Changelog].
 
 [Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
 
+## [0.4.0] - 2019-01-01
+- Support of I/O statistics for RocksDb.
```
Comment (ordian) [inline kvdb-rocksdb/CHANGELOG.md]: could you add links to the PRs as well
```suggestion
- Support of I/O statistics for RocksDB. (#294)[https://github.com/paritytech/parity-common/pull/294]
```

File: parity-util-mem/CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 
-The format is based on [Keep a Changelog]. 
+The format is based on [Keep a Changelog].
 
 [Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
 
+## [0.4.0] - 2020-01-01
+- Added implementation of `MallocSizeOf` for non-std `hashbrown::HashMap` and `lru::LRUMap`. [#293](https://github.com/paritytech/parity-common/pull/293)
```
Comment (ordian) [inline parity-util-mem/CHANGELOG.md]: I think #300 is also worth mentioning (Added support for tuples with length up to 12)

Comment (NikVolf) [reply parity-util-mem/CHANGELOG.md]: More like an implementation detail to me, but ok :)

File: kvdb-rocksdb/CHANGELOG.md [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
 
-The format is based on [Keep a Changelog]. 
+The format is based on [Keep a Changelog].
 
 [Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
 
+## [0.4.0] - 2019-01-01
+- Support of I/O statistics for RocksDB. [#294](https://github.com/paritytech/parity-common/pull/294)
```
Comment (dvdplm) [inline kvdb-rocksdb/CHANGELOG.md]:
```suggestion
- Add I/O statistics for RocksDB. [#294](https://github.com/paritytech/parity-common/pull/294)
```

**Comment by NikVolf** [comment]
> Nice.
> 
> Maybe dumb question but are you sure we need breaking releases for all the crates? Aren't most changes added features (as opposed to changed behaviour)?

1. `kvdb-rocksdb` depends on `parity-utils-mem`, the latter has to be updated with major bump, to the former as well.
2. kvdb::KeyValueDB now should implement MallocSizeOf, which is breaking change, it also propagates to other crates.
3. Some implementation details might produce performance degradation and so we might want an option to downgrade without pain.

**Comment by NikVolf** [comment]
Would be good also to include #302

File: primitive-types/CHANGELOG.md [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 
-The format is based on [Keep a Changelog]. 
+The format is based on [Keep a Changelog].
 
 [Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
 
+## [0.6.2] - 2019-01-03
+- Expose to_hex and from_hex from impl-serde. [#302]((https://github.com/paritytech/parity-common/pull/239))
```
Comment (ordian) [inline primitive-types/CHANGELOG.md]:
```suggestion
- Expose to_hex and from_hex from impl-serde. [#302](https://github.com/paritytech/parity-common/pull/302)
```

Comment (NikVolf) [reply primitive-types/CHANGELOG.md]: ops sorry :)

File: kvdb/CHANGELOG.md [hunk 5]
```diff
@@ -1,11 +1,14 @@
...
-The format is based on [Keep a Changelog]. 
+The format is based on [Keep a Changelog].
 
 [Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
 
+## [0.3.0] - 2020-01-03
+- I/O statistics API. [#294](https://github.com/paritytech/parity-common/pull/294)
+
```
Comment (ordian) [inline kvdb/CHANGELOG.md]:
```suggestion
- Removed `KeyValueDBHandler` trait. [#304](https://github.com/paritytech/parity-common/pull/304)

```

File: kvdb-memorydb/CHANGELOG.md [hunk 6]
```diff
@@ -1,11 +1,14 @@
...
 
-The format is based on [Keep a Changelog]. 
+The format is based on [Keep a Changelog].
 
 [Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
 
+## [0.3.0] - 2019-01-03
+- InMemory key-value database now can report memory used (via `MallocSizeOf`).
```
Comment (ordian) [inline kvdb-memorydb/CHANGELOG.md]:
```suggestion
- InMemory key-value database now can report memory used (via `MallocSizeOf`). [#292](https://github.com/paritytech/parity-common/pull/292)

```

**Pull request merged**